test/uk/gov/hmrc/minorentityidentificationfrontend/services/ValidateTrustKnownFactsServiceSpec.scala

```diff
@@ -93,13 +93,67 @@ class ValidateTrustKnownFactsServiceSpec extends AnyWordSpec with Matchers with
         mockRetrieveTrustKnownFactsConnector.retrieveTrustKnownFacts(testSautr) returns Future.successful(None)
         mockStorageService.storeIdentifiersMatch(testJourneyId, DetailsNotFound) returns Future.successful(SuccessfullyStored)
 
-        val result = await(TestJourneyService.validateTrustKnownFacts(testJourneyId, testSautr, Some("AB0 0AA")))
+        val result = await(TestJourneyService.validateTrustKnownFacts(journeyId = testJourneyId,
+          optSaUtr = Some(testSautr),
+          optSaPostcode = Some("AB0 0AA"),
+          optCHRN = Some(testCharityHMRCReferenceNumber))
+        )
 
         result mustBe DetailsNotFound
 
         mockStorageService.storeIdentifiersMatch(testJourneyId, DetailsNotFound) was called
       }
     }
+
+    "return UnMatchableWithoutRetry" when {
+      "the user provides no Sautr but provides CHRN (in this case postcode does not matter)" in {
+
+        List(None, Some(testSaPostcode)).map(optSaPostCode => {
+
+          mockStorageService.storeIdentifiersMatch(testJourneyId, UnMatchableWithoutRetry) returns Future.successful(SuccessfullyStored)
+
+          val result = await(TestJourneyService.validateTrustKnownFacts(journeyId = testJourneyId,
+            optSaUtr = None,
+            optSaPostcode = optSaPostCode,
+            optCHRN = Some(testCharityHMRCReferenceNumber))
+          )
+
+          result mustBe UnMatchableWithoutRetry
+
+          mockStorageService.storeIdentifiersMatch(testJourneyId, UnMatchableWithoutRetry) was called
+
+          mockStorageService.storeTrustsKnownFacts(testJourneyId, testTrustKnownFactsResponse) wasNever called
+
+          reset(mockStorageService)
+
+        })
+
+      }
+    }
+
+    "return UnMatchableWithRetry" when {
+      "the user provides no Sautr and no CHRN (in this case postcode does not matter)" in {
+        List(None, Some(testSaPostcode)).map(optSaPostCode => {
```

Comment:
Same as above about the postcode. There would not be a postcode in this scenario so only ned the test for postcode = None

Reply:
okay